plugins/inputs/docker/docker.go

```diff
@@ -303,6 +324,10 @@ func (d *Docker) gatherInfo(acc telegraf.Accumulator) error {
 	d.engine_host = info.Name
 	d.serverVersion = info.ServerVersion
 
+	if d.SwarmReporting == "leader" {
+		d.nodeID = info.Swarm.NodeID
```

Comment:
I think we should return this ID, and then we can pass the `leaderID` back into `gatherSwarmInfo`.  This can be done regardless of the setting but double check that it works if there is no swarm leader, I worry that info.Swarm may be unset in this case.  This will help ensure that these operations are done in the correct order.

Reply:
Good idea - I also figured it wouldn't be populated, which is why I put the check in there, but it's probably better to know what to expect and handle it appropriately.